tests/cpp_api_tests/rocAL_audio_unittests/rocAL_audio_unittests.cpp

```diff
@@ -0,0 +1,184 @@
+    if ((matchedIndices == frames) && matchedIndices != 0) {
+        pass_status = true;
+    }
+
+    return pass_status;
+}
+
+int test(int test_case, const char *path, const char *ref_path, int downmix, int gpu);
+int main(int argc, const char **argv) {
+    // check command-line usage
+    const int MIN_ARG_COUNT = 2;
+    printf("Usage: ./rocal_audio_unittests <audio-dataset-folder> <test_case> <downmix> <device-gpu=1/cpu=0> \n");
+    if (argc < MIN_ARG_COUNT)
+        return -1;
+
+    int argIdx = 0;
+    const char *path = argv[++argIdx];
+    const char *ref_path = nullptr;
+    unsigned test_case = 0;
+    bool downmix = false;
+    bool gpu = 0;
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        ref_path = argv[++argIdx];
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        test_case = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        downmix = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        gpu = atoi(argv[++argIdx]);
+    
+    if (gpu) {  // TODO - Will be removed when GPU support is added for Audio pipeline
+        std::cerr << "WRN : Currently Audio unit test supports only HOST backend\n";
+        gpu = false;
+    }
+
+    int return_val = test(test_case, path, ref_path, downmix, gpu);
+    return return_val;
+}
+
+int test(int test_case, const char *path, const char *ref_path, int downmix, int gpu) {
+    int inputBatchSize = 1;
+    std::cout << ">>> test case " << test_case << std::endl;
+    std::cout << ">>> Running on " << (gpu ? "GPU" : "CPU") << std::endl;
+
+    auto handle = rocalCreate(inputBatchSize,
+                              gpu ? RocalProcessMode::ROCAL_PROCESS_GPU : RocalProcessMode::ROCAL_PROCESS_CPU, 0,
+                              1);
+
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Could not create the Rocal contex\n";
+        return -1;
+    }
+
+    std::cout << ">>>>>>> Running AUDIO DECODER" << std::endl;
+    rocalAudioFileSourceSingleShard(handle, path, 0, 1, true, false, false, downmix);
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Audio source could not initialize : " << rocalGetErrorMessage(handle) << std::endl;
+        return -1;
+    }
+
+    // Calling the API to verify and build the augmentation graph
+    rocalVerify(handle);
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Could not verify the augmentation graph " << rocalGetErrorMessage(handle);
+        return -1;
+    }
+
+    int iteration = 0;
+    float *buffer = nullptr;
+    int frames = 0;
+    high_resolution_clock::time_point t1 = high_resolution_clock::now();
+    while (rocalGetRemainingImages(handle) >= static_cast<size_t>(inputBatchSize)) {
+        std::cout << "\n Iteration:: " << iteration<<"\n";
+        iteration++;
+        if (rocalRun(handle) != 0) {
+            break;
+        }
+        RocalTensorList output_tensor_list = rocalGetOutputTensors(handle);
```

Comment:
I think we can store the output for only the last run. 
Move the rocalGetOutputTensors out of this loop. So only last entry is saved.